bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/SolaxLocalAccessHandler.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solax.internal;
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.solax.internal.connectivity.LocalHttpConnector;
+import org.openhab.binding.solax.internal.connectivity.rawdata.LocalConnectRawDataBean;
+import org.openhab.binding.solax.internal.model.InverterData;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SolaxLocalAccessHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Konstantin Polihronov - Initial contribution
+ */
+@NonNullByDefault
+public class SolaxLocalAccessHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SolaxLocalAccessHandler.class);
+
+    private static final int INITIAL_SCHEDULE_DELAY_SECONDS = 5;
+
+    private SolaxConfiguration config;
+
+    private LocalHttpConnector localHttpConnector;
+
+    private @Nullable ScheduledFuture<?> schedule;
+
+    public SolaxLocalAccessHandler(Thing thing) {
+        super(thing);
+
+        config = getConfigAs(SolaxConfiguration.class);
```

Comment:
Do you need the configuration as instance variable, or would it be sufficient to keep it as a local variable in `initialize()`? I'm not sure, but I think you are only using the three parameters in this method.

Reply:
I need the password and the hostname to instantiate the localHttpConnector which is used to retrieve the data. If I move the config as local variable, then the same warning appears for the localHttpConnector. Maybe I can instantiate the localHttpConnector and set the data in the initialize() method but it's ugly ... very ugly. :) ... or I can say that the field localHttpconnector is nullable and handle it properly wherever is used (one place in the code as far as i see...) What do you think?

Reply:
You need to create `localHttpConnector` in `initialize()` as well since otherwise configuration changes still wouldn't have any affect. So yes, you would have to make it `@Nullable` or `@NonNullByDefault({})` to avoid warnings when we know the lifecycle but the analyzer doesn't.